lib/gollum-lib/markups.rb

```diff
@@ -6,7 +6,7 @@ class Markup
         Kramdown::Document.new(content, :auto_ids => false, :input => "markdown").to_html
     }
 
-    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/)
+    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/, :h1 => /^(#\s+)(\w+.*)/)
```

Comment:
@dometto 
I see where you're heading with matching two groups (whitespaces and content in this case) and substituting like this `#{$1}GOLLUMSRCMP#{i}GOLLUMSRCMP#{$2}`. **Smart!** However, I think there's a slightly better solution because we're only interested in `real-section-number => source-line-or-direct-offset` mappings. The resulting substituted raw data will not be presented and thus we're free to completely substitute the original header's content, as long as the new content is alphanumeric!

So, this should be a simpler (and a bit more markup-independent) solution:
* Matching - `if line =~ /^#(.*)$/`
* Substitution - `line[$1] = "GOLLUMSRCMP#{i}GOLLUMSRCMP"`

Since we're completely substituting the original content, `GOLLUMSRCMP#{i}GOLLUMSRCMP` might even turn to `#{i}`!

**NOTES:**
* Is this possible and reliable for all markups? I think header syntax for some markups is multi-line by definition...
* Maybe we should substitute the headers for direct character offsets (start position of the header syntax)? After all, that is what we want and what's best to use when substituting the section for a new version, later on.